### ISSUE TITLE
Disable multiple tag names for free-form tag formats

### ISSUE THREAD

**Issue by euri10** [beetbox/beets#350]
I don't know if this comes from beets or if it's me configuring ncmpcpp poorly.
I got this line in .ncmpcpp/config :
song_columns_list_format = "(5)[red]{n}(5f)[green]{l} (25)[cyan]{a} (40)[]{t|f} (25)[red]{b}"
This is the default way to display the songs in the column format in ncmpcpp except that I added the track number: (5)[red]{n}
The issue is that tracks are displayed that way :
![track](https://f.cloud.github.com/assets/1104190/871594/4f367d4e-f84f-11e2-9f86-0efad2f1c4bd.PNG)
If I go into the ncmpcpp tag editor :
![tags](https://f.cloud.github.com/assets/1104190/871601/85cfa13c-f84f-11e2-97ec-1d129b90fb9d.PNG)
Of course all those tracks were imported with beets,
it seems like a strange format. Is that a bug from beets mistagging the tracks or a ncmcpp rendering issue ?

**Comment by dennishamester** [comment]
This really bugged me. I wrote a patch that fixed this in ncmpcpp: arybczak/ncmpcpp#54

**Comment by sampsyo** [comment]
Awesome! Here's hoping that gets merged.

**Comment by jreinert** [comment]
i tried my luck contributing to mpd and fixing the issue "at the root" but it didn't go particularily well... https://github.com/MaxKellermann/MPD/pull/2

**Comment by sampsyo** [comment]
Wow, what a ridiculously hostile developer. :cry:

What about, for example, just using the first tag found? I guess I just don't understand why MPD would ever want to show multiple track tags, even if they're different.

**Comment by dennishamester** [comment]
I tried something similar some time ago:
http://thread.gmane.org/gmane.comp.audio.musicpd.devel/3377
https://github.com/dennishamester/MPD/commit/cbb5d2d9a51110d6b322d7dd4f50a8c19289f273

But there was no response from the developer at all. Since then I've given up getting this upstream.

**Comment by sampsyo** [comment]
Wow; that's really depressing. I suppose this is just another instance of different open-source projects trying to blame each other. :confused: 

I guess we're left with trying to invent a config option for beets: `compatible_tags: no` or some such thing.

**Comment by carnager** [comment]
maybe a pure musicbrainz config option :smile:

**Comment by carnager** [comment]
> What about, for example, just using the first tag found? I guess I just don't understand why MPD would ever want to show multiple track tags, even if they're different.

this would not be a good solution, since many users use multiple artist tags for collaborations.

**Comment by sampsyo** [comment]
@carnager, what do you mean by "pure MusicBrainz"?

**Comment by carnager** [comment]
@sampsyo https://picard.musicbrainz.org/docs/mappings/ (might be outdated)

**Comment by sampsyo** [comment]
Ah, so exactly matching what Picard in particular uses. I'm not sure if that exactly replication is really necessary—to fix the issue at hand, all we need to do is stop writing multiple versions of a tag for compatibility's sake.

**Comment by sentriz** [comment]
ah sorry @sampsyo - I never found this issue when I searched.

hopefully either mpd, ncmpcpp, or beets comes up with something - it would be great to see this fixed.  
when I open up one of the tracks that has the problem in foobar2000, it shows two fields: "tracknumber" and "trackc" - both with the same value, just like ncmpcpp.
that was after I rewrote my tags with beets. is it just beets and foobar2000, ncmpcpp and others not agreeing on a standard?

**Comment by sampsyo** [comment]
No problem!

Here's a brief summary of the current state of things:
- Tools disagree about which of the two names `tracknumber` and `trackc` should be used for this data.
- So, to maximize compatibility, beets (really, MediaFile) uses a "read any, write all" policy.
- This works great for 99% of software, which reads one or the other but doesn't care if both are set.
- ncmpcpp is the 1% that reads *and displays* both fields, even when they hold the same value.

We have two bad options:
- Status quo: ncmpcpp breaks. (And it sounds like the ncmpcpp developers are hostile toward any changes in this regard.)
- Remove one or the other: we break unknown software that has been silently working fine so far.

I think the only option for us is to add a configuration option specifically for people who want to use ncmpcpp.

**Comment by dennishamester** [comment]
Just to clarify, this is _not_ a pure ncmpcpp issue. MPD reads the files and stores the same tracknumber twice. ncmpcpp just happens to be the client that displays two tracknumbers iff mpd sends two or more. I also never got the impression that the ncmpcpp maintainers are hostile towards changes. The MPD maintainer however, doesn't even want to discuss the matter.

My suggestion is to go with a new configuration option in beets, even if this seems more like a workaround than an actual solution.

**Comment by sampsyo** [comment]
Yes, thanks for clarifying -- and my apologies for remembering wrong; it was the MPD developer who shut down the conversation, not the ncmpcpp people at all.

**Comment by carnager** [comment]
to be fair, mpd stores tags in a list, if there are more than one of a type, so its pretty easy for clients to filter out if there are duplicates in the list.

**Comment by sentriz** [comment]
Would it be possible to stop writing one of the two tags with a plugin like Zero? or does beets consider both `trackc` and `tracknumber` the same tag?

**Comment by sampsyo** [comment]
Alas, no—it's just as you suspected; zero sees them as one logical tag.

**Comment by ArchangeGabriel** [comment]
Hey there, new here but definitively affected by this and wanted to share my thoughts on it:

* According to me and my understanding of thing, mpd has the right behaviour. Since it provides tag editing interface for client, mpd should show all existing tags (unless this interface is accessed differently than the playing one?). What can be argued however is that ncmpcpp should only show one when multiple are presents in the playing mode, while displaying them all in the editing mode. Fun fact is I didn’t even tried ncmpcpp on my beets collection yet, thought this is in fact my current player, but came here because of my second point.

* This is not just and mpd (world) issue: this is an issue for anyone editing tags with other things than beets, they would look at files cluttered by numerous similar tags (and that’s my case, because beets can’t — or at least I don’t know how — fill all my needs, so I also use ExFalso aside). Or even just look at them with things like ffprobe or the like. And you can’t say or convince me that this is not the right behaviour for these tools, yet you might not want to agree with beets maximum compatibility policy and so not having extra tags for those things.

So, yes, please add a `strict_tags` option or a way to disable some tags to be written (since this can’t be done in current zero plugin). This is actually a blocking issue for me.

**Comment by mineo** [comment]
> Since it provides tag editing interface for client, mpd should show all existing tags

MPD doesn't provide a tag editing interface, ncmpcpp does that via taglib - that's why you have to configure `mpd_music_dir` in ncmpcpp for the tag editor to work.

**Comment by ArchangeGabriel** [comment]
Hum, OK. So must do my other MPD clients I suppose then. Then I suppose MPD should not care about non strict tags too or only consider one for each type, but we know this is not going to happen (could in ncmpcpp though).

Still, my second (and main for me) point is still valid.

**Comment by sampsyo** [comment]
Any help to add a mode like this would be greatly appreciated!

**Comment by ArchangeGabriel** [comment]
Where in the code does beets/MediaFile write tags?

**Comment by sampsyo** [comment]
Well, that's pretty much the job of MediaFile—it's our library for reading and writing tags. Specifically, you might want to start with the `store` methods.

**Comment by dinojr** [comment]
@jreinert 
I'm trying to use your fork of MaxKellerman/MPD but it won't build: I get 

I'm tried reporting the issue at your github fork but it seems you have disabled the issue tracking.

**Comment by jreinert** [comment]
@dinojr i completely forgot about my MPD fork. I merged the upstream master into my fork and pushed the changes. It builds at least.

**Comment by jtpavlock** [comment]
See [https://github.com/arybczak/ncmpcpp/commit/6ebf00eb5d3e6a66ad567aa09ca575f5e0c9a982](this ncmpccp commit) for those just worried about ncmpcpp.

**Comment by sampsyo** [comment]
Great news!

**Comment by carnager** [comment]
yeah for relaying an actual issue to downstream...
